Gruntfile.js

```diff
@@ -0,0 +1,145 @@
+module.exports = function(grunt) {
+    grunt.initConfig({
+        concurrent: {
+            serverAndTest: ["nodemon", "jasmine-node"],
+            options: {
+                logConcurrentOutput: true
+            }
+        },
+        nodemon: {
+            dev: {
+                // NEVER PUT A PATH IN THIS FILE NAME
+                script: "server.js",
+                options: {
+                    // PUT PATH HERE INSTEAD
+                    // THIS TOOK ME 4 HOURS TO FIX
+                    cwd: __dirname + "/server"
+                }
+            }
+        },
+        karma: {
+            continuous: {
+                configFile: "config/karma.conf.js",
+                singleRun: true
+            }
+        }
+    });
+    
+    //
+    // npm Tasks
+    //
+    grunt.loadNpmTasks("grunt-concurrent");
+    grunt.loadNpmTasks("grunt-nodemon");
+    grunt.loadNpmTasks("grunt-karma");
+    
+    //
+    // db-clear
+    //
+    grunt.registerTask("db-clear", "Clears the database of all data.", function() {
+        var mongoose = require("mongoose");
+        var dbConfig = require("./config/db");
+        
+        var done = this.async();
+        
+        mongoose.connect(dbConfig.url);
+        var connection = mongoose.connection;
+        
+        connection.once("open", function() {
+            connection.db.dropDatabase(function(err) {
+                if(err) {
+                    grunt.log.writeln("Error: " + err);
+                }
+                else {
+                    grunt.log.writeln("Successfully dropped development database.");
+                }
+                
+                connection.close(done);
+            });
+        });
+    });
+    
+    //
+    // db-populate
+    //
+    grunt.registerTask("db-populate", "Populates the database with dummy data.", function() {
+        var mongoose = require("mongoose");
+        var dbConfig = require("./config/db");
+        var Contact = require("./server/models/contact").Contact;
+        
+        var dummydata = grunt.file.readJSON("./config/development-data.json").allContacts;
+        var done = this.async();
+        
+        mongoose.Promise = global.Promise;
```

Comment:
I don't think this is necessary anymore since I'm no longer using promises in this task.